default/AI/ShipDesignAI.py

```diff
@@ -1924,6 +1910,77 @@ def _starting_guess(self, available_parts, num_slots):
             ret_val[-1] = num_slots - 1
         return ret_val
 
+
+def _add_shipdesign_fo(design_name, hull, parts_list, model="fighter", description="", icon="", name_desc_in_string_table=False, verbose=False):
+    """This is basically a wrapper around fo.issueCreateShipDesignOrder to also update the cache
+
+    :param design_name: the name of the design
+    :type design_name: str
+    :param description: a human readable description of the design
+    :type description: str
+    :param hull: the name of the hull to use
```

Comment:
> Well, if you want to be very strict, it's actually a key string, which gets looked up in the stringtables to retrieve the actual name of the hull (that is presented to the player).

No, if you want to be very strict, then what is presented to the player is some completely arbitrary, random and meaningless string linked to the hull, i.e. a flavor text.
The name of the hull in any part of the code is by the very definition the string that ```hull::name()``` returns or from a python perspective what ```hull.name``` returns (which happens to be the same function call...).